client/client_test.go

```diff
@@ -3763,14 +3763,44 @@ func TestGetAllTargetInfo(t *testing.T) {
 	require.Len(t, targetSignatureData, 7)
 
 	makeSureWeHitEachCase = make(map[string]struct{})
+	checkSignatures(
+		t,
+		targetSignatureData,
+		makeSureWeHitEachCase,
+		repo,
+		*targetsCurrentTarget,
+		*targetsLatestTarget,
+		*level1CurrentTarget,
+		*level1OtherTarget,
+		*level2CurrentTarget,
+		*level2Level2Target,
+		*level1Level2Level2Target,
+		key1,
+		key2,
+		key3,
+	)
+	require.Len(t, makeSureWeHitEachCase, 7)
+
+	// nonexistent targets
+	targetSignatureData, err = repo.GetAllTargetMetadataByName("level23")
+	require.Error(t, err)
+	require.Nil(t, targetSignatureData)
+	targetSignatureData, err = repo.GetAllTargetMetadataByName("invalid")
+	require.Error(t, err)
+	require.Nil(t, targetSignatureData)
+}
+
+func checkSignatures(t *testing.T, targetSignatureData []TargetSignedStruct, makeSureWeHitEachCase map[string]struct{}, repo *NotaryRepository,
```

Comment:
I hate how this came out so happy to change it. Will bug you IRL to help me see how to make it better 👍